openapi.yaml

```
@@ -1980,6 +1980,292 @@ paths:
+        Create a new secret in this organization.
+
+        <Note>Only organization admins can create secrets.</Note>
+      parameters:
+        - $ref: '#/components/parameters/ib_context'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                alias:
+                  type: string
+                  description: Name of the secret.
+                description:
+                  type: string
+                  description: Description of the secret.
+                value:
+                  type: string
+                  description: Content of the secret.
+                allowed_workspaces_type:
+                  type: string
+                  description: Which workspaces in this organization are allowed to use this secret in custom functions.
+                  enum:
+                    - ALL
+                    - SOME
+                    - NONE
+                allowed_workspaces:
+                  type: array
+                  description: Which workspaces are allowed to use this secret, if only some are allowed.
+                  items:
+                    type: string
+              required:
+                - alias
+                - value
+                - allowed_workspaces_type
+      responses:
+        '200':
+          description: Secret created successfully.
+        default:
+          description: Error response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+      x-fern-examples:
+        - code-samples:
+            - sdk: curl
+              code: |
+                  curl -X POST "${API_ROOT}/v2/secrets" \
+                    -H "Authorization: Bearer ${API_TOKEN}" \
+                    -H "IB-Context: ${IB_CONTEXT}"\
+                    -H "Content-Type: application/json" \
+                    -d '{
+                          "alias": "my_secret",
+                          "description": "Sample secret",
+                          "value": "my password",
+                          "allowed_workspaces_type": "SOME",
+                          "allowed_workspaces": ["workspace1", "workspace2"]
+                        }'
+            - sdk: python
+              code: |
+                  # AI Hub Python SDK example
+                  from aihub import AIHub
+
+                  client = AIHub(api_root="<API-ROOT>",
+                                  api_key="<API-TOKEN>",
+                                  ib_context="<IB-CONTEXT>")
+
+                  result = client.secrets.create(
+                    alias="my_secret",
+                    description="Sample secret",
+                    value="my password",
+                    allowed_workspaces_type="SOME",
+                    allowed_workspaces=['workspace1', 'workspace2']) # Only required when allowed_workspaces_type="SOME"
```

```suggestion
                    # only required when allowed_workspaces_type=“SOME" 
                    allowed_workspaces=['workspace1', 'workspace2'])
```

To avoid horizontal scrolling in our tiny sample code window.